platforms: ihp-sg13g2: May Update

dnltz opened the pull request:
Picked @KrzysztofHerman update script and modified it slightly. Synced files afterwards with this script and added the bondpad manually.

Should we remove metal fill here since KLayout is supposed to do this task?

This PR is based on #2002 with some changes and addresses PR #2066

dnltz committed:
flow: platforms: ihp-sg13g2: Add script to update files

This script can download the latest Open PDK files and helps
to easily sync ORFS.

Signed-off-by: KrzysztofHerman <herman@ihp-microelectronics.com>
Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
flow: platforms: ihp-sg13g2: Sync files with 5a42d03

https://github.com/IHP-GmbH/IHP-Open-PDK/commit/5a42d03194e8c98558f4e34538338a60550f89b9

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
flow: platforms: ihp-sg13g2: Allow sg13g2_antennanp cell

Remove sg13g2_antennanp from DONT_USE_CELLS.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
flow: platforms: ihp-sg13g2: Add 70x70 bondpads

Add manually bondpads with 70x70 dimensions since they're not
supported in the Open PDK yet.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

maliberty commented on flow/platforms/ihp-sg13g2/fill.json:
```diff
@@ -1,103 +1,5 @@
-            }
-        }
-    },
+    "layers" : {},
```
If you do remove fill you should just remove the entire fill and update the config.

dnltz committed:
flow: platforms: ihp-sg13g2: Add layer map

The layer map is missing in sg13g2.lyt and layers won't be
drawn correctly.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
flow: Makefile: Replace .map files in do-klayout

IHP's PDK uses a map file to define layers. We need to alter
the path to point from the OBJECTS_DIR to the PDK dir. Add
a sed command after LEF modifications.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

maliberty commented on flow/platforms/ihp-sg13g2/lef/sg13g2_io_notracks.lef:
What is the difference between this and flow/platforms/ihp-sg13g2/lef/sg13g2_io.lef ?  Are both required?

dnltz replied:
@KrzysztofHerman can we drop the io_notrack LEF file here?

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
```diff
@@ -0,0 +1,78 @@
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/liberty/sg13g2_io_dummy.lib', 'lib')
+# GDS
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/gds/sg13g2_stdcell.gds', 'gds')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/gds/sg13g2_io.gds', 'gds')
```
Why is bondpad_70x70.gds missing?

dnltz replied:
bondpad is currently only located in this repo. I think the long-term solution is to generate those cells with Klayout similar to the sealring. So, just a temporarily solution to build chips.

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
```diff
@@ -0,0 +1,78 @@
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/lef/sg13g2_tech.lef', 'lef')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/lef/sg13g2_stdcell.lef', 'lef')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/lef/sg13g2_io.lef', 'lef')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/lef/sg13g2_io_notracks.lef', 'lef')
```
Why is bondpad_70x70.lef missing?

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
```diff
@@ -0,0 +1,78 @@
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/lib/sg13g2_stdcell_typ_1p20V_25C.lib', 'lib')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/lib/sg13g2_stdcell_typ_1p50V_25C.lib', 'lib')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/liberty/sg13g2_io_dummy.lib', 'lib')
+# GDS
```
The RM*gds are missing

dnltz replied:
Fixed that. Thanks!

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
```diff
@@ -0,0 +1,78 @@
+# GDS
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_stdcell/gds/sg13g2_stdcell.gds', 'gds')
+download_github_file('ihp-sg13g2/libs.ref/sg13g2_io/gds/sg13g2_io.gds', 'gds')
+# LEF
```
The RM*lef are missing

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
```diff
@@ -0,0 +1,78 @@
+download_github_file('ihp-sg13g2/libs.tech/klayout/tech/sg13g2.lyp')
+download_github_file('ihp-sg13g2/libs.tech/klayout/tech/sg13g2.lyt')
+download_github_file('ihp-sg13g2/libs.tech/klayout/tech/drc/sg13g2.lydrc', 'drc')
+# LIB
```
The RM*.lib are missing

maliberty commented on flow/platforms/ihp-sg13g2/sg13g2_update.py:
The RM*.v are missing from the script.  I also missed importing them so they should be added to this PR.
